js/net_test.ts

```diff
@@ -105,27 +126,27 @@ testPerm({ net: true }, async function netListenAsyncIterator(): Promise<void> {
   listener.close();
   conn.close();
 });
- */
 
-/* TODO Fix broken test.
-testPerm({ net: true }, async function netCloseReadSuccess() {
-  const addr = "127.0.0.1:4500";
+testPerm({ net: true }, async function netCloseReadSuccess(): Promise<void> {
+  const addr = createAddr();
   const listener = Deno.listen("tcp", addr);
   const closeDeferred = deferred();
   const closeReadDeferred = deferred();
-  listener.accept().then(async conn => {
-    await closeReadDeferred.promise;
-    await conn.write(new Uint8Array([1, 2, 3]));
-    const buf = new Uint8Array(1024);
-    const readResult = await conn.read(buf);
-    assertEquals(3, readResult.nread);
-    assertEquals(4, buf[0]);
-    assertEquals(5, buf[1]);
-    assertEquals(6, buf[2]);
-    conn.close();
-    closeDeferred.resolve();
-  });
-  const conn = await Deno.dial("tcp", addr);
+  listener.accept().then(
```

Comment:
On the CI this is where is breaks on a windows env. adding `.catch()=>{ console.log('catch');}` outputs:
```
test netCloseReadSuccess_permR0W0N1E0U0H0 Other: Listener has been closed
    at maybeError (gen/cli/bundle/main.js:5186:18)
    at handleAsyncMsgFromRust (gen/cli/bundle/main.js:6929:23)
    at handleAsyncMsgFromRust (shared_queue.js:157:9)readresultBIG CATCH... ok
test netDoubleCloseRead_permR0W0N1E0U0H0 Catch accept... FAILED
AssertionError
    at assert (C:/temp/deno/js/deps/https/deno.land/std/testing/asserts.ts:45:15)
    at netDoubleCloseRead (C:/temp/deno/js/net_test.ts:204:5)
    at async runTestsSerial (C:/temp/deno/js/deps/https/deno.land/std/testing/mod.ts:112:13)
    at async runTests (C:/temp/deno/js/deps/https/deno.land/std/testing/mod.ts:147:9)
    at async main (C:/temp/deno/js/deps/https/deno.land/std/testing/main.ts:5:5)
test netDoubleCloseWrite_permR0W0N1E0U0H0 ... FAILED
```

So seems like the test is ok BUT the next breaks due to it. Be cause of:
`assertEquals(err.kind, Deno.ErrorKind.NotConnected);`
Instead of NotConnected we have `ConnectionReset`

Reply:
`test netCloseReadSuccess_permR0W0N1E0U0H0 Other: Listener has been closed` error message shown later is caught error by former test (`testAsyncIterator`) .

Windows CI failed at the same test but got different error with you. I can't debug windows and no hint is in sight.

Reply:
It is the same error as me. I've just added a `.catch()` on the async iterator. That's why you see `Catch accept...`
I can try to debug it on windows. Maybe you got an idea where to start?
Is here is a good start: https://github.com/denoland/deno/pull/2269/files#diff-d68f31c6cc35294dfb936e465af774d9R197 ?

Reply:
Really? 
>Error: read error
    at read (gen/cli/bundle/main.js:6938:17)
    at async netCloseReadSuccess (C:/deno/js/net_test.ts:132:24)

Latest CI failed with error above on https://ci.appveyor.com/project/deno/deno/builds/24370690 . It looks different with `AssertionError`. Are there some  misunderstandings?

Reply:
I think so. But i'll try to investigate more on this.